src/Build/Evaluation/Evaluator.cs

```diff
@@ -1659,6 +1666,15 @@ private List<ProjectRootElement> ExpandAndLoadImports(string directoryOfImportin
             return allProjects;
         }
 
+        private static readonly string CouldNotResolveSdk = ResourceUtilities.GetResourceString("CouldNotResolveSdk");
+        private static readonly string ProjectImported = ResourceUtilities.GetResourceString("ProjectImported");
```

Comment:
Rather than hardcoding these, wouldn't it be better to just cache all resources in ResouceUitlities as they're asked for?

Reply:
Turns out they're already cached in the BCL so calling ResourceUtilities.GetResourceString is cheap enough (not as cheap as a static field access though).

Reply:
@cdmihai another aspect of this just occurred to me. Evaluator.cs is a huge perf bottleneck and a hot path. This is why we should strive to microoptimize here no matter big or small, when it is reasonable of course. Get all the background noise out of the way as much as possible.